pytest_httpserver/blocking_http_server.py

```diff
@@ -0,0 +1,155 @@
+from queue import Empty
+from queue import Queue
+from typing import Any
+from typing import Dict
+from typing import Mapping
+from typing import Optional
+from typing import Pattern
+from typing import Union
+
+from werkzeug.wrappers import Request
+from werkzeug.wrappers import Response
+
+from pytest_httpserver.httpserver import METHOD_ALL
+from pytest_httpserver.httpserver import UNDEFINED
+from pytest_httpserver.httpserver import HeaderValueMatcher
+from pytest_httpserver.httpserver import HttpServerBase
+from pytest_httpserver.httpserver import QueryMatcher
+from pytest_httpserver.httpserver import RequestHandlerBase
+from pytest_httpserver.httpserver import URIPattern
+
+
+class BlockingRequestHandler(RequestHandlerBase):
+    """
+    Provides responding to a request synchronously.
+
+    This class should only be instantiated inside the implementation of the :py:class:`BlockingHttpServer`.
+    """
+
+    def __init__(self):
+        self.response_queue = Queue()
+
+    def respond_with_response(self, response: Response):
+        self.response_queue.put_nowait(response)
+
+
+class BlockingHttpServer(HttpServerBase):
```

Comment:
Sorry for nitpicking. :)

Could you rename `Http` to `HTTP` here, and in other classes?

We have `HTTPServer` class defined in `httpserver.py` and I think this would be unified in that case.

I think this would affect all the classes having `Http` instead of `HTTP`.

There would be a good argument about which one looks better, maybe I would also chose `Http` if I could start this from the beginning but having it unified is better I think.

Reply:
I chose the `Http` because it is the standard python recommendation.
We can use the `HTTP`, if you prefer that.